### ISSUE TITLE
🐛 --colors option broken

### ISSUE THREAD

**Issue by hyperknot** [biomejs/biome#3824]
### Environment information

```block
pnpm biome rage
CLI:
  Version:                      1.8.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.17.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.9.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

--colors option is broken in CLI 

1. Autodetect doesn't detect piped output with `| cat`, it still sends color characters.
2. `--colors=off` doesn't turn colors off (might be the actual reason behind 1.)

Currently the output cannot be grepped correctly, because of all the decoration characters beint sent to grep, etc.

### Expected result

Colors should turn off either with `| cat` or with `--colors=off`

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
Please provide a reproduction, or instructions to reproduce it on our machines

**Comment by hyperknot** [comment]
1. `pnpm i @biomejs/biome`
2. `pnpm biome check --write --unsafe --colors=off --log-level=info --log-kind=pretty . | cat`

Colors are still present

**Comment by difosfor** [comment]
I wanted to just see a list of files where errors were found so I turned to grep, only to find that did not seem to have any effect. After some testing with the latest version (2.2.2) I found that only the last three lines of output were affected by grep; see example below. I did find --colors=off to result in Biome no longer showing colors, but I did not find Biome to automatically switch off the colors when piping to grep nor grep working even without the colors.

Am I doing something wrong? Is this a known issue? Is there a workaround?

Example the output of `$ npx biome lint --max-diagnostics=1 --colors=off`:

```
src/engine-dash/loader/controllers/RepresentationStream.ts:36:1 lint/style/useExportsLast ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  i All exports should be declared after all non-export statements.
  
     34 │ }
     35 │ 
   > 36 │ export class RepresentationStream extends EventEmitter<RepresentationStreamEvents> {
        │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   > 37 │   readonly log: Logger;
         ...
  > 249 │   }
  > 250 │ }
        │ ^
    251 │ 
    252 │ function getNextTimeline(
  
  i Move this statement before the export statements to keep all exports at the end of the module.
  

The number of diagnostics exceeds the limit allowed. Use --max-diagnostics to increase it.
Diagnostics not shown: 394.
Checked 160 files in 369ms. No fixes applied.
Found 10 warnings.
```

Versus the output of `$ npx biome lint --max-diagnostics=1 --colors=off | grep foo`:

```
src/engine-dash/loader/controllers/RepresentationStream.ts:36:1 lint/style/useExportsLast ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  i All exports should be declared after all non-export statements.
  
     34 │ }
     35 │ 
   > 36 │ export class RepresentationStream extends EventEmitter<RepresentationStreamEvents> {
        │ ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   > 37 │   readonly log: Logger;
         ...
  > 249 │   }
  > 250 │ }
        │ ^
    251 │ 
    252 │ function getNextTimeline(
  
  i Move this statement before the export statements to keep all exports at the end of the module.

```

PS: I also found that --log-kind=compact makes no difference from the default or explicit --log-kind=pretty

**Comment by siketyan** [comment]
@difosfor That's because Biome emits those diagnostics to the standard error instead of standard output. You can redirect them to stdout:

```sh
biome check 2>&1 | grep ...
```

**Comment by difosfor** [comment]
Ah, thanks, that works!

I didn't think of that; I was expecting biome to output to stdout while it's just doing it's linting job and I didn't give it any incorrect arguments or something like that.
